README
NAME
    Dist::Zilla::Plugin::Catalyst - set of plugins for working with Catalyst

VERSION
    version 0.16

SYNOPSIS
    for quick start

            dzil new -P Catalyst <ProjectName>

DESCRIPTION
    The following is a list of plugins in this distribution to help you
    integrate Catalyst and Dist::Zilla

    *   Dist::Zilla::Plugin::MinterProfile::Catalyst Default Minting profile

    *   Dist::Zilla::Plugin::Catalyst::New Create a new Catalyst Project

PATCHES
    Please read the SubmittingPatches file included with this Distribution.
    Patches that are of sufficient quality, within the goals of the project
    and pass the checklist will probably be accepted.

BUGS
    Please report any bugs or feature requests on the bugtracker website
    https://github.com/xenoterracide/Dist-Zilla-Plugin-Catalyst/issues

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHORS
    *   Caleb Cushing <xenoterracide@gmail.com>

    *   Tomas Doran <bobtfish@bobtfish.net>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2011 by Caleb Cushing.

    This is free software, licensed under:

      The Artistic License 2.0 (GPL Compatible)



Changes
Revision history for Dist::Zilla::Plugin::Catalyst

0.16      2011-08-07

0.15      2011-02-17
	- plan does not accept a hash, fix syntax error

0.14      2011-02-16
	- Create 01app.t
	- rewrite changes to match CPAN::Changes::Spec
	- rewrite tests to be more comprehensive, and more DRY
	- Enable MetaJSON
	- explicitly depend on Perl 5.6

0.13      2011-01-16
	- fix bug where Catalyst::Helper::mk_dir was attempting to create dirs in
	  the location of the profile.ini for
	  Dist::Zilla::MintingProfile::Catalyst which tends to be in a system
	  owned directory.

0.12      2011-01-08
	- skip tests on devel moose

0.11      2011-01-08
	- add MinterProfile::Catalyst for quickstart

0.10      2011-01-05
	- skip executable tests on windows

0.09      2011-01-03
	- move to use Path::Class for OS independant code

0.08      2010-12-12
	- API: Rename DZ::Plugin::Catalyst::Helper to DZ::Catalyst::Helper. It's
	  not really a Plugin, but a subclass of Catalyst::Helper.

0.07      2010-12-11
	- Add minting tests to ensure cli functionality, and correct file creation

0.06      2010-07-26
	- don't create pod tests which have seperate functions in newer versions
	  of Catalyst::Devel

0.05      2010-07-22
	- No longer output the Catalyst templated Changes file

0.04      2010-07-22
	- add support to get author from config.ini

0.03      2010-07-22
	- fix perms for script/* should have been executable

0.02      2010-06-29
	- Updated SubmittingPatches doc and moved it's location. re-releasing to
	  attempt to fix file conflict bug

0.01      2010-06-27
	- Initial release, patches and comments welcome


t/lib/DZPCshared.pm
use strict;
use warnings;
package DZPCshared;
use Path::Class;
use Moose;
use namespace::autoclean;

has 'appname' => (
	is       => 'ro',
	required => 1,
);

has 'tempdir' => (
	is       => 'ro',
	required => 1,
);

has 'directories' => (
	isa      => 'ArrayRef[Path::Class::Dir]',
	traits   => ['Array'],
	is       => 'ro',
	required => 1,
	lazy     => 1,
	default  => sub {
		my $self = shift;
		my $mr   = dir( $self->tempdir )->subdir('mint');
		my $mrl  = $mr->subdir('lib');
		my $mrr  = $mr->subdir('root');
		my $mrs  = $mr->subdir('script');
		my $mrt  = $mr->subdir('t');
		my $mrri = $mr->subdir('root')->subdir('static')->subdir('images');
		return my $directories = [
			$mr,
			$mrl,
			$mrr,
			$mrs,
			$mrt,
			$mrri,
		];
	},
);

has 'files' => (
	isa      => 'ArrayRef[Path::Class::File]',
	traits   => ['Array'],
	is       => 'ro',
	required => 1,
	lazy     => 1,
	default  => sub {
		my $self = shift;
		my ( $mr, $mrl, $mrr, $mrs, $mrt, $mrri ) = @{ $self->directories };
		my $lc_app = lc $self->appname;
		return my $files = [
			$mr->file   ( $lc_app . '.conf'               ),
			$mrl->file  ( $self->appname . '.pm'          ),
			$mrl->subdir( $self->appname )->subdir('Controller')->file('Root.pm'),
			$mrr->file  ( 'favicon.ico'                   ),
			$mrri->file ( 'btn_120x50_built.png'          ),
			$mrri->file ( 'btn_120x50_built_shadow.png'   ),
			$mrri->file ( 'btn_120x50_powered.png'        ),
			$mrri->file ( 'btn_120x50_powered_shadow.png' ),
			$mrri->file ( 'btn_88x31_built.png'           ),
			$mrri->file ( 'btn_88x31_built_shadow.png'    ),
			$mrri->file ( 'btn_88x31_powered.png'         ),
			$mrri->file ( 'btn_88x31_powered_shadow.png'  ),
			$mrri->file ( 'catalyst_logo.png'             ),
			$mrt->file  ( '01app.t'                       ),
		];
	},
);

has 'scripts' => (
	isa      => 'ArrayRef[Path::Class::File]',
	traits   => ['Array'],
	is       => 'ro',
	required => 1,
	lazy     => 1,
	default  => sub {
		my $self = shift;
		my ( $mr, $mrl, $mrr, $mrs, $mrt, $mrri ) = @{ $self->directories };
		my $lc_app = lc $self->appname;
		return my $scripts = [
			$mrs->file  ( $lc_app . '_cgi.pl'     ),
			$mrs->file  ( $lc_app . '_create.pl'  ),
			$mrs->file  ( $lc_app . '_fastcgi.pl' ),
			$mrs->file  ( $lc_app . '_server.pl'  ),
			$mrs->file  ( $lc_app . '_test.pl'    ),
		];
	},
);

has 'files_not_created' => (
	isa      => 'ArrayRef[Path::Class::File]',
	traits   => ['Array'],
	is       => 'ro',
	required => 1,
	lazy     => 1,
	default  => sub {
		my $self = shift;
		my ( $mr, $mrl, $mrr, $mrs, $mrt, $mrri ) = @{ $self->directories };
		my $lc_app = lc $self->appname;
		return my $files_not_created = [
			$mr->file  ( 'README'          ),
			$mr->file  ( 'Changes'         ),
			$mr->file  ( 'Makefile.PL'     ),
			$mrt->file ( '02pod.t'         ),
			$mrt->file ( '03podcoverage.t' ),
		];
	},
);

__PACKAGE__->meta->make_immutable;
1;


Makefile.PL

use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
install_share module => "Dist::Zilla::MintingProfile::Catalyst", "profiles";


my %WriteMakefileArgs = (
  'ABSTRACT' => 'set of plugins for working with Catalyst',
  'AUTHOR' => 'Caleb Cushing <xenoterracide@gmail.com>, Tomas Doran <bobtfish@bobtfish.net>',
  'BUILD_REQUIRES' => {
    'Dist::Zilla::Tester' => '0',
    'English' => '0',
    'File::Find' => '0',
    'File::Temp' => '0',
    'FindBin' => '0',
    'Test::More' => '0'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.30',
    'File::ShareDir::Install' => '0.03'
  },
  'DISTNAME' => 'Dist-Zilla-Plugin-Catalyst',
  'EXE_FILES' => [],
  'LICENSE' => 'artistic_2',
  'NAME' => 'Dist::Zilla::Plugin::Catalyst',
  'PREREQ_PM' => {
    'Catalyst::Helper' => '1.30',
    'Dist::Zilla::File::FromCode' => '0',
    'Dist::Zilla::File::InMemory' => '0',
    'Dist::Zilla::Role::MintingProfile::ShareDir' => '0',
    'Dist::Zilla::Role::ModuleMaker' => '0',
    'Moose' => '0',
    'Path::Class' => '0',
    'namespace::autoclean' => '0'
  },
  'VERSION' => '0.16',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

package
MY;
use File::ShareDir::Install qw(postamble);


